Sources/Kitura/staticFileServer/StaticFileServer.swift

```diff
@@ -101,9 +102,25 @@ open class StaticFileServer: RouterMiddleware {
     /// - Parameter customResponseHeadersSetter: an object of a class that
     /// implements `ResponseHeadersSetter` protocol providing a custom method to set
     /// the headers of the response.
-    public init(path: String = "./public", options: Options = Options(),
+    public init?(path: String = "./public", options: Options = Options(),
                  customResponseHeadersSetter: ResponseHeadersSetter? = nil) {
-        absoluteRootPath = StaticFileServer.ResourcePathHandler.getAbsolutePath(for: path)
+        let rootPathAbsolute = StaticFileServer.ResourcePathHandler.getAbsolutePath(for: path)
+        absoluteRootPath = rootPathAbsolute
+        // Check the supplied path is a directory and if not fail initialisation
+        var isDirectory = ObjCBool(false)
+
+        let pathExists = FileManager.default.fileExists(atPath: absoluteRootPath, isDirectory: &isDirectory)
+        #if !os(Linux) || swift(>=4.1)
+        let isDirectoryBool = isDirectory.boolValue
+        #else
+        let isDirectoryBool = isDirectory
+        #endif
+        if !pathExists {
+            Log.warning("StaticFileServer being initialised with invalid path: \(rootPathAbsolute)")
+        } else if !isDirectoryBool {
+            Log.error("StaticFileServer cannot be intialised with a file")
```

Comment:
If the initializer is no longer failable, this becomes 'should not be initialized with a file'